docs/fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md

```diff
@@ -119,6 +132,123 @@ using var fileStream = new FileStream("MyAssembly.exe", FileMode.Create, FileAcc
+                    metadataRootBuilder: new MetadataRootBuilder(metadataBuilder),
+                    ilStream: ilStream,
+                    debugDirectoryBuilder: debugDirectoryBuilder,
+                    entryPoint: entryPointHandle);
+
+    BlobBuilder peBlob = new BlobBuilder();
+    peBuilder.Serialize(peBlob);
+
+    using var fileStream = new FileStream("MyAssembly.exe", FileMode.Create, FileAccess.Write);
+    peBlob.WriteContentTo(fileStream);
+}
+
+static DebugDirectoryBuilder GeneratePDB(MetadataBuilder pdbBuilder, ImmutableArray<int> rowCounts, MethodDefinitionHandle entryPointHandle)
+{
+    BlobBuilder portablePdbBlob = new BlobBuilder();
+    PortablePdbBuilder portablePdbBuilder = new PortablePdbBuilder(pdbBuilder, rowCounts, entryPointHandle);
+    BlobContentId pdbContentId = portablePdbBuilder.Serialize(portablePdbBlob);
+    // In case saving PDB to a file
+    using FileStream fileStream = new FileStream("MyAssemblyEmbeddedSource.pdb", FileMode.Create, FileAccess.Write);
+    portablePdbBlob.WriteContentTo(fileStream);
+
+    DebugDirectoryBuilder debugDirectoryBuilder = new DebugDirectoryBuilder();
+    debugDirectoryBuilder.AddCodeViewEntry("MyAssemblyEmbeddedSource.pdb", pdbContentId, portablePdbBuilder.FormatVersion);
+    // In case embedded in PE:
+    // debugDirectoryBuilder.AddEmbeddedPortablePdbEntry(portablePdbBlob, portablePdbBuilder.FormatVersion);
+    return debugDirectoryBuilder;
+}
+```
+
+Further user could add CustomDebugInformation by calling the AddCustomDebugInformation method from the `pdbBuilder` instance to add source embedding and source indexing etc. advanced PDB info.
+
+```csharp
+private static void EmbedSource(MetadataBuilder pdbBuilder)
+{
+    byte[] sourceBytes = File.ReadAllBytes("MySourceFile2.cs");
+    BlobBuilder sourceBlob = new BlobBuilder();
+    sourceBlob.WriteBytes(sourceBytes);
+    pdbBuilder.AddCustomDebugInformation(MetadataTokens.DocumentHandle(1),
+        pdbBuilder.GetOrAddGuid(new Guid("0E8A571B-6926-466E-B4AD-8AB04611F5FE")), pdbBuilder.GetOrAddBlob(sourceBlob));
+}
+```
+
+### Adding resources with PersistedAssemblyBuilder
+
+With `MetadataBuilder.AddManifestResource(...)` you could add as many resources as needed, but streams have to be concatenated together into one `BlobBuilder` that you pass into `ManagedPEBuilder` argument.
+Following example shows how to create resources and attach it to the assembly created:
+
+```csharp
+public static void SetResource()
+{
+    PersistedAssemblyBuilder ab = CreateAssemblyAndItsContents();
+    MetadataBuilder metadata = ab.GenerateMetadata(out BlobBuilder ilStream, out _);
+
+    using MemoryStream stream = new MemoryStream();
+    ResourceWriter myResourceWriter = new ResourceWriter(stream);
+    myResourceWriter.AddResource("AddResource 1", "First added resource");
+    myResourceWriter.AddResource("AddResource 2", "Second added resource");
+    myResourceWriter.AddResource("AddResource 3", "Third added resource");
```

Comment:
```suggestion
    myResourceWriter.AddResource("String 1", "First string");
    myResourceWriter.AddResource("String 2", "Second string");
    myResourceWriter.AddResource("String 3", "Third string");
```
I think this looks a bit better - matches https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcewriter?view=net-8.0#examples